### ISSUE TITLE
Acceptance criteria for merging ethereum-master into master

### ISSUE THREAD

**Issue by drwasho** [OpenBazaar/openbazaar-go#1707]
Note: this issue will be edited on an ongoing basis.

**Our goal is to merge `ethereum-master` into `master` when the server can stably deliver Ethereum functionality (i.e. support for transactions and orders), at the required number precision level, while supporting backwards compatible functionality for non-ETH coins and prior-version nodes.**

With this goal in mind, we would like to achieve this goal before DEVCON so an update can be pushed to Haven users in time for the conference.

Presently, `ethereum-master` is waiting on two branches to be merged in:

1. PR #1675 repo.Listing
2. PR #TBD Ethereum wallet

During the code review of PR #1675, a number of issues were identified that require further attention and push into our deadline.

So, to achieve this goal in the desired timeframe, we will need to take the following approach:

1. Identify PR-merge blocking issues to fix for #1675 repo.Listing
2. Review and identify PR-merge blocking issues for PR #TBD Ethereum wallet
3. Itemise issues raised by @placer14 and @cpacia for both PRs 
4. Triage issues identified in (3) into 2 buckets: 
    1. External-facing issues => affects the client and user's ability to use OpenBazaar.
    2. Internal-facing issues => sub-optimal code to be improved that doesn't affect the client or user in a direct way.
5. Prioritise resolving external facing issues via end-to-end testing before addressing internal ones, which can started before and after ethereum-master is merged into master.

### End-to-end testing

This is an umbrella term that covers:

1. **QA testing.** This process isn't automated yet, and needs the test suite to be triggered manually.
2. **Functional testing.** This is covers testing an endpoint or scenario (i.e. similar to QA tests) manually.
